### ISSUE TITLE
LiTE/Image: CreateImageProvider() failed

### ISSUE THREAD

**Issue by Frieren360** [directfb2/DirectFB2#153]
Anytime I try to run an example that uses LiTE (like df_andi and lite_*) I get the error:
`(!) DirectFBError [LiTE/Image: CreateImageProvider() failed]: A general or unknown error occured`
I do not get the error with other examples like df_fire or df_knuckles and they run fine. I am running Void Linux with musl. This is the output of dfbinfo:
```
Video Modes
   2256x1504-32
   1920x1200-32
   1680x1050-32
   1280x1024-32
   1440x900-32
   1280x800-32
   1024x768-32
   800x600-32
   640x480-32

Screen (00) DRMKMS Screen                           (primary screen)
   Caps: MIXERS ENCODERS OUTPUTS 
   Mixer (0) DRMKMS Mixer
     Caps:               FULL 
     Layers (full mode): (00) (01) (02) (03) (04) (05) (06) (07) (08) (09) 

   Encoder (0) DRMKMS Encoder
     Type:               DIGITAL 
     Caps:               FREQUENCY RESOLUTION 
     Output Resolutions: 640_480 800_600 1024_768 1280_720 1280_1024 1600_1200 1920_1080 1920_1200 
     Mixer:              0

   Output (0) DRMKMS Output
     Caps:               RESOLUTION 
     Resolutions:        640_480 800_600 1024_768 1280_720 1280_1024 1600_1200 1920_1080 1920_1200 
     Signals:            
     Encoder:            0

     Layer (00) DRMKMS Primary Layer                (primary layer)
        Type:    GRAPHICS 
        Caps:    SURFACE 

     Layer (01) DRMKMS Plane Layer 1              
        Type:    GRAPHICS 
        Caps:    SURFACE ALPHACHANNEL LEVELS SCREEN_POSITION 

     Layer (02) DRMKMS Plane Layer 2              
        Type:    GRAPHICS 
        Caps:    SURFACE ALPHACHANNEL LEVELS SCREEN_POSITION 

     Layer (03) DRMKMS Plane Layer 3              
        Type:    GRAPHICS 
        Caps:    SURFACE ALPHACHANNEL LEVELS SCREEN_POSITION 

     Layer (04) DRMKMS Plane Layer 4              
        Type:    GRAPHICS 
        Caps:    SURFACE OPACITY ALPHACHANNEL LEVELS SCREEN_POSITION 

     Layer (05) DRMKMS Plane Layer 5              
        Type:    GRAPHICS 
        Caps:    SURFACE OPACITY ALPHACHANNEL LEVELS SCREEN_POSITION 

     Layer (06) DRMKMS Plane Layer 6              
        Type:    GRAPHICS 
        Caps:    SURFACE ALPHACHANNEL LEVELS SCREEN_POSITION 

     Layer (07) DRMKMS Plane Layer 7              
        Type:    GRAPHICS 
        Caps:    SURFACE ALPHACHANNEL LEVELS SCREEN_POSITION 

     Layer (08) DRMKMS Plane Layer 8              
        Type:    GRAPHICS 
        Caps:    SURFACE ALPHACHANNEL LEVELS SCREEN_POSITION 

     Layer (09) DRMKMS Plane Layer 9              
        Type:    GRAPHICS 
        Caps:    SURFACE ALPHACHANNEL LEVELS SCREEN_POSITION 

Input Device (00) AT Translated Set 2 keyboard      (primary keyboard)
   Vendor  ID: 0x0001
   Product ID: 0x0001
   Type:       KEYBOARD 
   Caps:       KEYS 
Input Device (03) FRMW0004:00 32AC:0006 Consumer    (primary remote control)
   Vendor  ID: 0x32ac
   Product ID: 0x0006
   Type:       REMOTE 
   Caps:       KEYS AXES 
Input Device (01) PIXA3854:00 093A:0274 Mouse       (primary mouse)
   Vendor  ID: 0x093a
   Product ID: 0x0274
   Type:       MOUSE 
   Caps:       AXES BUTTONS 
Input Device (10) PIXA3854:00 093A:0274 Touchpad  
   Vendor  ID: 0x093a
   Product ID: 0x0274
   Type:       MOUSE 
   Caps:       AXES BUTTONS 
Input Device (11) Virtual Input                   
   Vendor  ID: 0x0000
   Product ID: 0x0000
   Type:       KEYBOARD MOUSE JOYSTICK REMOTE VIRTUAL 
   Caps:       KEYS AXES BUTTONS
   ```

**Comment by caramelli** [comment]
df_andi does not depend on LiTE, so I assume you get the following error when running df_andi:

       (!) DirectFBError [buffer->CreateImageProvider( buffer, &provider )]: A general or unknown error occurred

What is the output of the following command:

	strace df_andi 2>&1 | grep IDirectFBImageProvider

**Comment by Frieren360** [comment]
```
open("/usr/local/lib/directfb-2.0-0/interfaces/IDirectFBImageProvider", O_RDONLY|O_CLOEXEC|O_DIRECTORY) = 27
open("/usr/local/lib/directfb-2.0-0/interfaces/IDirectFBImageProvider/libidirectfbimageprovider_webp.so", O_RDONLY|O_CLOEXEC) = 28
open("/usr/local/lib/directfb-2.0-0/interfaces/IDirectFBImageProvider/libidirectfbimageprovider_jpeg.so", O_RDONLY|O_CLOEXEC) = 28
open("/usr/local/lib/directfb-2.0-0/interfaces/IDirectFBImageProvider/libidirectfbimageprovider_bmp.so", O_RDONLY|O_CLOEXEC) = 28
open("/usr/local/lib/directfb-2.0-0/interfaces/IDirectFBImageProvider/libidirectfbimageprovider_imlib2.so", O_RDONLY|O_CLOEXEC) = 28
open("/usr/local/lib/directfb-2.0-0/interfaces/IDirectFBImageProvider/libidirectfbimageprovider_gif.so", O_RDONLY|O_CLOEXEC) = 28
open("/usr/local/lib/directfb-2.0-0/interfaces/IDirectFBImageProvider/libidirectfbimageprovider_heif.so", O_RDONLY|O_CLOEXEC) = 28
open("/usr/local/lib/directfb-2.0-0/interfaces/IDirectFBImageProvider/libidirectfbimageprovider_png.so", O_RDONLY|O_CLOEXEC) = 28
open("/usr/local/lib/directfb-2.0-0/interfaces/IDirectFBImageProvider/libidirectfbimageprovider_yuv.so", O_RDONLY|O_CLOEXEC) = 28
open("/usr/local/lib/directfb-2.0-0/interfaces/IDirectFBImageProvider/libidirectfbimageprovider_jxl.so", O_RDONLY|O_CLOEXEC) = 28
open("/usr/local/lib/directfb-2.0-0/interfaces/IDirectFBImageProvider/libidirectfbimageprovider_tiff.so", O_RDONLY|O_CLOEXEC) = 28
open("/usr/local/lib/directfb-2.0-0/interfaces/IDirectFBImageProvider/libidirectfbimageprovider_jasper.so", O_RDONLY|O_CLOEXEC) = 28
open("/usr/local/lib/directfb-2.0-0/interfaces/IDirectFBImageProvider/libidirectfbimageprovider_gdkpixbuf.so", O_RDONLY|O_CLOEXEC) = 28
writev(2, [{iov_base="", iov_len=0}, {iov_base="(*) Direct/Interface: Loaded 'Gd"..., iov_len=84}], 2(*) Direct/Interface: Loaded 'GdkPixbuf' implementation of 'IDirectFBImageProvider'
```

**Comment by caramelli** [comment]
OK, in your DirectFB2 directory, the DFIFF (DirectFB Fast Image File Format) image provider should have been built and should be installed in /usr/local/lib/directfb-2.0-0/interfaces/IDirectFBImageProvider

You can just put it back manually:

	install build/interfaces/IDirectFBImageProvider/libidirectfbimageprovider_dfiff.so /usr/local/lib/directfb-2.0-0/interfaces/IDirectFBImageProvider

Or to be sure all DirectFB2 files are installed correctly, simply run:

	meson install -C build/

**Comment by Frieren360** [comment]
I have tried to install it manually and recompile and reinstall DirectFB2 but I still get the same error. I can confirm the fille `/usr/local/lib/directfb-2.0-0/interfaces/IDirectFBImageProvider/libidirectfbimageprovider_dfiff.so` exists. `strace` does not show the file being opened either. I have managed to workaround the error by renaming or creating a symlink of `libidirectfbimageprovider_dfiff.so` to a different file image provider format like `libidirectfbimageprovider_tiff.so`.
